### PR TITLE
Add support for contains operator for authz header matcher

### LINKED ISSUE #32173: Add support for contains authz header matcher
It would be very helpful to be able to use the `HeaderMatcher_ContainsMatch` so that we don't have to rely on the sort order of an array header value. For example if we use `X-Forwarded-Client-Cert` to match on a SAN of the cert  and the order of the names changes the authorization policies have to be updated.

Happy to push a PR if you think that this is a good change.

### PR DESCRIPTION
**Pull request by dmorina**
closes #32173

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by dmorina** [comment]
@googlebot I signed it!

File: pilot/pkg/security/authz/matcher/header.go [hunk 1]
```diff
@@ -32,6 +32,13 @@ func HeaderMatcher(k, v string) *routepb.HeaderMatcher {
 				PresentMatch: true,
 			},
 		}
+	} else if strings.HasPrefix(v, "*") && strings.HasSuffix(v, "*") {
+		return &routepb.HeaderMatcher{
+			Name: k,
+			HeaderMatchSpecifier: &routepb.HeaderMatcher_ContainsMatch{
+				ContainsMatch: v[1 : len(v)-1],
```
Comment (kebe7jun) [inline pilot/pkg/security/authz/matcher/header.go:39]:
```suggestion
				ContainsMatch: v[1:len(v)-1],
```

**Review by kebe7jun — COMMENTED** [review]
/ok-to-test

File: pilot/pkg/security/authz/matcher/header.go (hunk 1 above)
Comment (kebe7jun) [reply pilot/pkg/security/authz/matcher/header.go]: oh, It looks like I was wrong.

Comment (dmorina) [reply pilot/pkg/security/authz/matcher/header.go:39]: no worries, reverted it

**Review by yangminzhu — CHANGES_REQUESTED** [review]
Thanks for the change! I think this is a reasonable improvement but we will need to hold the PR a little bit because it involves API changes even the change itself is rather small.

Please note the authorization API is currently in beta stage and we have high requirements for any API changes in a beta feature.

Could you bring this up in the next security WG meeting ([schedule](https://docs.google.com/document/d/12Xz8fCuql2tPL-cpdLXGGE44kYT4o9NB6mljSiS_01U/edit#)) and discuss the API change there? Once the API change is approved, we can go ahead with the implementation.

cc @lizan @liminw @myidpt

**Comment by dmorina** [comment]
Sounds good! I will join the 4/28 meeting to discuss this change. Thanks!

**Comment by dmorina** [comment]
@yangminzhu we discussed this in the last WG meeting 4/28 and the change was approved, let me know how you want to proceed from here. Thanks!

**Pull request closed without merging**